src/test/scala/com/microsoft/hyperspace/index/DeltaLakeIntegrationTest.scala

```diff
@@ -310,6 +321,262 @@ class DeltaLakeIntegrationTest extends QueryTest with HyperspaceSuite {
+
+          // The index should be applied for the updated version.
+          checkExpectedIndexUsed(None, "deltaIndex/v__=1")
+          checkExpectedIndexUsed(Some(0), "deltaIndex/v__=0")
+
+          appendData(doRefresh = false) // delta version 3
+          appendData(doRefresh = false) // delta version 4
+          appendData(doRefresh = false) // delta version 5
+          appendData(doRefresh = true) // delta version 6
+          appendData(doRefresh = false) // delta version 7
+          appendData(doRefresh = true) // delta version 8
+
+          val dataPathStr = new Path(dataPath).toString
+
+          checkExpectedIndexUsed(None, "deltaIndex/v__=3")
+          checkExpectedIndexUsed(Some(1), "deltaIndex/v__=1")
+          checkExpectedIndexUsed(Some(3), "deltaIndex/v__=1", dataPathStr)
+          checkExpectedIndexUsed(Some(5), "deltaIndex/v__=2")
+          checkExpectedIndexUsed(Some(6), "deltaIndex/v__=2")
+          checkExpectedIndexUsed(Some(7), "deltaIndex/v__=3")
+        }
+      }
+    }
+  }
+
+  test("Verify time travel query works well with incremental refresh & optimizeIndex.") {
+    withTempPathAsString { dataPath =>
+      import spark.implicits._
+      val dfFromSample = sampleData.toDF("Date", "RGUID", "Query", "imprs", "clicks")
+      dfFromSample.write.format("delta").save(dataPath)
+
+      val timestampMap = mutable.Map[Long, String]()
+      timestampMap.put(0, getSparkFormattedTimestamps(System.currentTimeMillis).head)
+
+      val indexName = "deltaIndex2"
+      val deltaDf = spark.read.format("delta").load(dataPath)
+
+      def appendData(doRefresh: Boolean): Unit = {
+        // Append data.
+        dfFromSample
+          .limit(3)
+          .write
+          .format("delta")
+          .mode("append")
+          .save(dataPath)
+
+        // Get the timestamp string for time travel query using "timestampAsOf" option.
+        val timestampStr = getSparkFormattedTimestamps(System.currentTimeMillis).head
+        // Sleep 1 second because the unit of commit timestamp is second.
+        Thread.sleep(1000)
+
+        if (doRefresh) {
+          hyperspace.refreshIndex(indexName, mode = "incremental")
+        }
+
+        val newDf = spark.read.format("delta").load(dataPath)
+        timestampMap.put(getDeltaLakeTableVersion(newDf), timestampStr)
+      }
+
+      def query(version: Option[Long] = None, asTimestamp: Boolean = false): DataFrame = {
+        val deltaDf = if (version.isDefined) {
+          if (asTimestamp) {
+            spark.read
+              .format("delta")
+              .option("timestampAsOf", timestampMap(version.get))
+              .load(dataPath)
+          } else {
+            spark.read.format("delta").option("versionAsOf", version.get).load(dataPath)
+          }
+        } else {
+          spark.read.format("delta").load(dataPath)
+        }
+        deltaDf.filter(deltaDf("clicks") <= 2000).select(deltaDf("query"))
+      }
+
+      def checkExpectedIndexUsed(
```

Comment:
Is it possible to merge with previous `checkExpectedIndexUsed` and test both path and version? (then you can just have one function for validation?)